### ISSUE TITLE
Splinter Cell Chaos Theory and Splinter Cell Double Agent do not launch

### ISSUE THREAD

**Issue by CDSFounder** [NVIDIAGameWorks/rtx-remix#37]
Splinter Cell Chaos Theory single player and Splinter Cell Double Agent single player (aka splintercell3 and splintercell4) do not launch when attempting to use RTX Remix. The game will sit on a black or white screen, but never load any UI or viewport. Appears to hang the game as no other logging or components of the game will load. Eventually Windows will indicate the game has stopped responding. Attempted both windowed mode and full screen.

- Which Remix version: Attempted version 0.1.1
-OS: Windows 11
- GPU: RTX 4090
- GPU driver version:  31.0.15.3129
- Which game build: Latest patched versions of the single player Splinter Cell Chaos Theory and Splinter Cell Double Agent.

Logs from Splinter Cell Chaos Theory: 
[d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11218617/d3d9.log)
[splintercell3_d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11218619/splintercell3_d3d9.log)

Logs from Splinter Cell Double Agent:
[d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11218630/d3d9.log)
[SplinterCell4_d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11218631/SplinterCell4_d3d9.log
[NvRemixBridge.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11218954/NvRemixBridge.log)

The log in both games appears to repeat the following lines:
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state D3DRS_DEBUGMONITORTOKEN
warn:  Message channel UWM_REMIX_BRIDGE_REGISTER_THREADPROC_MSG handshake timeout. Retrying...

**Comment by Drommedhar** [comment]
For me they launch, but the window is either really small or the game is running with sub 1fps (even in the menu).
The game running like garbage is the default for me most of the time with the resolution of the display being broken completly and also not able to change resolution etc. Looks pretty broken to me right now.

**Comment by Joshhhuaaa** [comment]
Using this PC configuration below, I was able to progress further into the game. Menus would render and I could even get into levels. The overlay would work and I was able to tell RTX Remix which textures were considered "UI" textures, which made it so the "Briefing" screen before levels properly rendered.

It is worth mentioning that Splinter Cell: Chaos Theory supports Shader Model versions 1.1, 2.0, and 3.0. However, Splinter Cell: Double Agent only supports Shader Model 3.0, so that game is likely going to be unbootable for the near future.

Both games are running under DirectX9.0c, and even though Chaos Theory seems to be running fine when using Shader Model 1.1 mode, it doesn't appear to do any path tracing. The image looks similar to the default game, minus a few graphical bugs.

While it may take some time to see both Chaos Theory and Double Agent function under RTX Remix with path tracing, the series provides an excellent showcase for this project.

RTX Remix Version: 0.1.1
OS: Windows 10, Version 22H2
GPU: RTX 2070 Super
GPU Driver Version: 531.41
Game: Splinter Cell: Chaos Theory (Latest Version: v1.05)

Here are some log files from a session that may be of help (starting a level, playing for a few minutes, then closing the game):

Logs from Splinter Cell: Chaos Theory:
[SplinterCell3_d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11236586/SplinterCell3_d3d9.log)
[d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11236587/d3d9.log)

**Comment by Drommedhar** [comment]
I could not even play the game once past the briefing, as it went down to 1fps instantly and stayed there.
Also it did not apply resolution most of the time.

**Comment by Joshhhuaaa** [comment]
> I could not even play the game once past the briefing, as it went down to 1fps instantly and stayed there. Also it did not apply resolution most of the time.

Yeah, I know what you are talking about, it seems to ignore screen resolutions and plays in a small 640x480 box. This issue seems to happen in Splinter Cell 1-4, as well as their Versus modes, possibly related to Unreal.

A temporary workaround I have found is to use ThirteenAG's widescreen patch for Splinter Cell: Chaos Theory. However, make sure to not overwrite the D3D9.dll that RTX Remix uses.

**Comment by Drommedhar** [comment]
Using the WidescreenFix really helped with the window stuff. I can run the game (at least the menus) now in the correct resolution.

Once I'm in game however, I'm still back to 1fps with completly scrambled visuals:
![image](https://user-images.githubusercontent.com/11786314/232209809-a40ab64e-422a-4e4b-9035-20edae17d5bd.png)
Looks like most of the textures are drawn semi transparent. Don't know why this is happening. I even tried the alpha blending options in the dev settings, with no luck.
And as you said, no paytracing or lighting stuff is doing something here, which might be related to the transparent textures I'm seeing.

**Comment by ColinM9991** [comment]
> Using this PC configuration below

What configuration? I don't see anything in your comment.

I ask as I'm also seeing the following message in my logs and unable to launch Chaos Theory with rtx-remix.
> warn: Message channel UWM_REMIX_BRIDGE_REGISTER_THREADPROC_MSG handshake timeout. Retrying...

**Comment by NV-LL** [comment]
Hello! Will you please check on the latest release and let us know if this issue persists? Thank you!

**Comment by CDSFounder** [comment]
> Hello! Will you please check on the latest release and let us know if this issue persists? Thank you!

I checked again. Now, with the experimental option "Capture vertices from Shader" enabled, the game renders like normal, however it does not appear to have RTX applying. If you go to the debug view, and try to view the geometry, you do not see any rainbows - the rendering still just looks like normal. 

However, if you disable "capture vertices from Shader" option, the game renders only a black screen. 

Does this indicate progress @NV-LL ?

**Comment by OndrejSzekely** [comment]
Hello @CDSFounder @nv-ajaus, any progress on this? I tried it yesterday and having the same problem. I tried to disable or lower all graphical options in the ini file, but no difference. :/

**Comment by NV-LL** [comment]
Unfortunately, we suspect these two titles are incompatible with Remix for now. We apologize for any frustration that may cause. At this time we're going to close this ticket.
